lib/checkstl.cpp

```diff
@@ -2016,6 +2016,38 @@ static bool accumulateBool(const Token * tok, unsigned int varid)
     return false;
 }
 
+static bool hasVarIds(const Token * tok, unsigned int var1, unsigned int var2)
+{
+    if(tok->astOperand1()->varId() == tok->astOperand2()->varId())
+        return false;
+    unsigned int varids[] = {var1, var2};
+    unsigned int * varidsEnd = varids+2;
+    if(std::find(varids, varidsEnd, tok->astOperand1()->varId()) == varidsEnd)
+        return false;
+    if(std::find(varids, varidsEnd, tok->astOperand2()->varId()) == varidsEnd)
+        return false;
+    return true;
+}
+
+static std::string minmaxCompare(const Token * condTok, unsigned int loopVar, unsigned int assignVar, bool invert=false)
+{
+    static const std::vector<std::string> minmaxAlgos = {
```

Comment:
This can be a plain old array. Please make it non-static to avoid thread safety problems.
I suggest a enum variable instead of the `int algo`.

Reply:
> I suggest a enum variable

I am not sure how that can help. The integer is used so i can easily flip the result. Thats not so easy with an enum. I might as well use the strings directly in that case.